hazelcast-jet-core/src/main/java/com/hazelcast/jet/processor/Sources.java

```diff
@@ -52,28 +55,72 @@ private Sources() {
      * {@code localParallelism * clusterSize}, otherwise some processors will have
      * no partitions assigned to them.
      * <p>
-     * If the underlying map is concurrently being modified, there are no guarantees
-     * given with respect to missing or duplicate items.
+     * Iterating the map should be done only when the {@link com.hazelcast.core.IMap} is not being
+     * mutated and the cluster is stable (there are no migrations or membership changes).
+     * In other cases, the iterator may not return some entries or may return an entry twice.
      */
     @Nonnull
     public static ProcessorMetaSupplier readMap(@Nonnull String mapName) {
         return ReadWithPartitionIteratorP.readMap(mapName);
     }
 
     /**
+     * Returns a meta-supplier of processor that will fetch entries from the
+     * Hazelcast {@code IMap} with the specified name. Entries will be filtered
+     * according to the given predicate and the result of the projection will be
+     * emitted to downstream. The processors will only access data local to the
+     * member and, if {@code localParallelism} for the vertex is above one,
+     * processors will divide the labor within the member so that each one gets
+     * a subset of all local partitions to read.
+     * <p>
+     * The number of Hazelcast partitions should be configured to at least
+     * {@code localParallelism * clusterSize}, otherwise some processors will have
+     * no partitions assigned to them.
+     * <p>
+     * Iterating the map should be done only when the {@link com.hazelcast.core.IMap} is not being
+     * mutated and the cluster is stable (there are no migrations or membership changes).
+     * In other cases, the iterator may not return some entries or may return an entry twice.
+     */
+    @Nonnull
+    public static <K, V, T> ProcessorMetaSupplier readMap(@Nonnull String mapName,
+                                                          @Nonnull DistributedFunction<Map.Entry<K, V>, T> projectionF,
```

Comment:
I would prefer the argument order to be predicate and then projection. Because the predicate is applied first, and then the projection.